fuel-core/src/schema/contract.rs

```diff
@@ -113,4 +121,112 @@ impl ContractBalanceQuery {
             asset_id,
         })
     }
+
+    async fn contract_balances(
+        &self,
+        ctx: &Context<'_>,
+        filter: ContractBalanceFilterInput,
+        first: Option<i32>,
+        after: Option<String>,
+        last: Option<i32>,
+        before: Option<String>,
+    ) -> async_graphql::Result<Connection<AssetId, ContractBalance, EmptyFields, EmptyFields>> {
+        let db = ctx.data_unchecked::<Database>().clone();
+
+        query(
+            after,
+            before,
+            first,
+            last,
+            |after: Option<AssetId>, before: Option<AssetId>, first, last| async move {
+                let (records_to_fetch, direction) = if let Some(first) = first {
+                    (first, IterDirection::Forward)
+                } else if let Some(last) = last {
+                    (last, IterDirection::Reverse)
+                } else {
+                    (0, IterDirection::Forward)
+                };
+
+                let after = after.map(fuel_tx::AssetId::from);
+                let before = before.map(fuel_tx::AssetId::from);
+
+                let start;
+                let end;
+
+                if direction == IterDirection::Forward {
+                    start = after;
+                    end = before;
+                } else {
+                    start = before;
+                    end = after;
+                }
+                // impl Iterator<Item = ContractBalance>
+                let mut balances = db
+                    .contract_balances(filter.contract.into(), start, Some(IterDirection::Forward))
```

Comment:
You've hardcoded the iterator direction instead of using the direction calculated above. When using backwards pagination we need to tell rocksdb to start from the end of the table instead of the beginning.

See the graphql connection specs for more details: https://relay.dev/graphql/connections.htm#sec-Forward-pagination-arguments